frame/contracts/src/wasm/mod.rs

```diff
@@ -305,7 +326,21 @@ impl<T: Config> WasmBlob<T> {
 			if let Some(code_info) = existing {
 				ensure!(code_info.refcount == 0, <Error<T>>::CodeInUse);
 				ensure!(&code_info.owner == origin, BadOrigin);
-				T::Currency::unreserve(&code_info.owner, code_info.deposit);
+				let _ = T::Currency::release(
+					&HoldReason::StorageDepositReserve.into(),
+					&code_info.owner,
+					code_info.deposit,
+					BestEffort,
+				);
+
+				<Pallet<T>>::deposit_event(
+					vec![T::Hashing::hash_of(&code_info.owner)],
+					Event::StorageDepositReleased {
+						who: code_info.owner.clone(),
+						amount: code_info.deposit,
+					},
+				);
```

Comment:
The old `Event::Reserved` had an empty vec as a `topic`, not sure if we want to keep the old behaviour or it makes sense to add a topic. 
Happy to hear some thoughts on this